components/infobox/wikis/mobilelegends/infobox_hero_custom.lua

```diff
@@ -0,0 +1,180 @@
+	table.insert(widgets, Cell{name = 'Energy Regen', content = {_args.energyreg}})
+	table.insert(widgets, Cell{name = 'Attack Type', content = {_args.attacktype}})
+	table.insert(widgets, Cell{name = 'Attack Damage', content = {_args.damage}})
+	table.insert(widgets, Cell{name = 'Attack Speed', content = {_args.attackspeed}})
+	table.insert(widgets, Cell{name = 'Attack Range', content = {_args.attackrange}})
+	table.insert(widgets, Cell{name = 'Ability Power', content = {_args.ap}})
+	table.insert(widgets, Cell{name = 'Armor', content = {_args.armor}})
+	table.insert(widgets, Cell{name = 'Magic Resistance', content = {_args.magicresistance}})
+	table.insert(widgets, Cell{name = 'Movement Speed', content = {_args.movespeed}})
+	table.insert(widgets, Title{name = 'Esports Statistics'})
+
+	local stats = HeroWL.create({hero = _args.heroname or _pagename})
+	stats = mw.text.split(stats, ';')
+	local winPercentage = (tonumber(stats[1]) or 0) / ((tonumber(stats[1]) or 0) + (tonumber(stats[2]) or 1))
+	winPercentage = Math.round({winPercentage, 4}) * 100
+	local statsDisplay = (stats[1] or 0) .. 'W : ' .. (stats[2] or 0) .. 'L (' .. winPercentage .. '%)'
+	table.insert(widgets, Cell{name = 'Win Rate', content = {statsDisplay}})
+
+	return widgets
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'header' then
+		return {
+			Header{
+				name = _args.heroname,
+				subHeader = _args.title,
+				image = _args.image,
+				imageDefault = _args.default,
+				imageDark = _args.imagedark or _args.imagedarkmode,
+				imageDefaultDark = _args.defaultdark or _args.defaultdarkmode,
+			},
+		}
+	elseif id == 'caption' then
+		table.insert(widgets, Center{content = {_args.quote}})
+	elseif id == 'type' then
+		local breakDownContents = {}
+		local lane = _args.lane
+		if not String.isEmpty(lane) then
+			lane = '<b>Lane</b><br>' .. Template.safeExpand(_frame, 'Class icon', {lane}, '')
+			table.insert(breakDownContents, lane)
+		end
+		local primaryRole = _args.primaryrole
+		if not String.isEmpty(primaryRole) then
+			primaryRole = '<b>Primary Role</b><br>' .. Template.safeExpand(_frame, 'Class icon', {primaryRole}, '')
+			table.insert(breakDownContents, primaryRole)
+		end
+		local secondaryRole = _args.secondaryrole
+		if not String.isEmpty(secondaryRole) then
+			secondaryRole = '<b>Secondary Role</b><br>' .. Template.safeExpand(_frame, 'Class icon', {secondaryRole}, '')
+			table.insert(breakDownContents, secondaryRole)
+		end
+		return {
+			Breakdown{classes = {'infobox-center'}, content = breakDownContents},
+			Cell{name = 'Real Name', content = {_args.realname}},
+		}
+	elseif id == 'cost' then
+		local cost = ''
+		if not String.isEmpty(_args.costbp) then
+			cost = cost .. _args.costbp .. ' ' .. _BATTLE_POINTS_ICON
+		end
+		if not String.isEmpty(_args.costdia) then
+			if cost ~= '' then
+				cost = cost .. '&emsp;&ensp;'
+			end
+			cost = cost .. _args.costdia .. ' ' .. _DIAMONDS_ICON
+		end
```

Comment:
Put it in a table and use table.concat. The advantages isn't create when there's only two ifs, but if it expands to more, repeating line 137-139 in every single if would be not great.